packages/language-server/src/plugins/typescript/LSAndTSDocResolver.ts

```diff
@@ -116,11 +119,31 @@ export class LSAndTSDocResolver {
         lang: ts.LanguageService;
         userPreferences: ts.UserPreferences;
     }> {
-        const lang = await this.getLSForPath(document.getFilePath() || '');
+        const { tsDoc, lsContainer, userPreferences } = await this.getLSAndTSDocWorker(document);
+
+        return { tsDoc, lang: lsContainer.getService(), userPreferences };
+    }
+
+    /**
+     * Retrieves the LS for operations that don't need cross-files information.
+     * can save some time by not synchronizing languageService program
+     */
+    async getLsForSyntheticOperations(document: Document): Promise<{
```

Comment:
I created this as a separate method because it gave it a more meaningful name, so the code intention is clearer. And, well, tsserver also does that😆,